### ISSUE TITLE
Tuned Corrupting My Grub Bootloader?

### ISSUE THREAD

**Issue by VoxR4710** [redhat-performance/tuned#174]
So I kept having a problem where my dedi wouldn't come back online. By pure luck , I checked the grub2 directory ( i am bit noob ) and found, restored an rpmsave... eureka...
I didn't care why i just went on living until it happened again last night, and on inspection , I've found the only difference is TUNED stuff! 

I had suspected this app as it was the only new thing when the issue first occured, but the documentation had me convinced it was safe so I stupidly retained it.
To dev: come to my discord and pm me, Vox, if you want more details i.e. the additions made.
https://discord.gg/5gJmPkX

Apparently the bug is this https://bugzilla.redhat.com/show_bug.cgi?id=1509515
But in my opinion that's not ok , if I was the type and had the time I would go around telling everyone to beware of this app.. I could've reinstalled my whole OS for nothing because this :)
Loved everything else about it & thanks anyway but I'ma stay away for safety

**Comment by yarda** [comment]
I understand your frustration, but I am sorry we cannot fix bugs caused by unresponsive grubby downstream.

Fortunately the problem will go away with f30, due to BLS:
https://fedoraproject.org/wiki/Changes/BootLoaderSpecByDefault

BTW one reason the BLS will be adopted by default is to get rid of grubby.

Even if you are hit by this bug you do not need to reinstall. It's enough to boot from live medium, chroot and then run grub2-mkconfig - it will regenerate your grub.cfg.

**Comment by yarda** [comment]
You should be able to switch to BLS even on f29, the support is there, but it's not enabled by default.

**Comment by VoxR4710** [comment]
Well this the thing man apart from regenerate grub I don't know what you're talking about. 
Tuned seems to be marketed as something to make tuning linux easier for people who can't/don't want the hassle of doing it manually, so my point really is that you should probably at least have the app warn users who might be effected by this as to what might happen and how to fix it. Don't mind me bro I'm out forever just trying to help u

**Comment by yarda** [comment]
Tuned is important component in Red Hat Enterprise Linux regarding user friendly performance tuning, and probably similarly in the CentOS. Other distros are packaging it slowly as well. The problem doesn't manifest there because there is no grubby or there is fixed grubby in the distros. Even the grubby was fixed upstream, just the Fedora downstream has unpatched grubby for some reason which is unclear to me.

The bootloader tuning is crucial for performance tuning, we could disable it in Fedora downstream but it would affect people using BLS. Also I wouldn't like to divert in Fedora downstream from Tuned upstream.

I agree this problem even if Fedora specific should be documented at least.

**Comment by VoxR4710** [comment]
Yeah, I love the idea. I forgot to mention, i'm on CentOS latest. 
I guess if it was built in it definitely wouldn't screw up the grub cfg so that would be cool. I dont have time to figure out what exactly went wrong and help but I can paste you the additions Tuned made which stopped my server booting if you like.
All I can tell you is , I would have loved to see a prompt when I set my Tuned profile that said like "Warning: RHEL OS detected, you may need to modify your grub bootloader after activation to prevent boot issues" or something like that :) Good luck with it all bro
